### PR TITLE
Advanced Multiple Testing Solution via limit distribution. 

### PR DESCRIPTION
**Pull request by tikhomirovd**
It's duplicate PR from [here](https://github.com/sb-ai-lab/HypEx/pull/9). 

More information in the [milestone](https://github.com/sb-ai-lab/HypEx/milestone/2)

New PR was created cause need some changes in our side

### CODE REVIEW

File: hypex/ab_test/limit_distribution.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        return int(2 * variances * ((quantile_1 - quantile_2) / minimum_detectable_effect) ** 2) + 1
+    else:
+        iteration_size = 3000  # Количество итераций
+        if quantile_1 is None:
+            quantile_1 = quantile_of_marginal_distribution(num_samples=number_of_samples,
+                                                           quantile_level=1 - significance_level / number_of_samples,
+                                                           variances=variances,
+                                                           )  # набор квантилей предельного распределения
+        sample_sizes = []  # для размеров выборки
+        for sample_index in range(number_of_samples):
```
Comment (Dmatryus) [inline hypex/ab_test/limit_distribution.py:109]: It looks scary, but there are no ideas how to make it better 😂

File: hypex/ab_test/limit_distribution.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+print(f'Sample size = {sample_size}')
+
+# Testing samples with equal conversion rate
+print('\nSamples with equal conversion rate')
+for _ in range(5):
+    samples = bernoulli.rvs(assumed_conversion, size=[num_samples, sample_size], random_state=random_state)
+    hypothesis = test_on_marginal_distribution(samples, significance_level=significance_level)
+    print(f'\tAccepted hypothesis H({hypothesis})')
+
+print("kek")
```
Comment (Dmatryus) [inline hypex/ab_test/limit_distribution.py:161]: Kek?)

File: hypex/ab_test/limit_distribution.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
+# Решаем задачу множественного тестирования
```
Comment (VasilievaKA) [inline hypex/ab_test/limit_distribution.py:1]: Docs in russian

File: hypex/ab_test/limit_distribution.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+                                sample_size / (variances[sample_index] + variances[i]))
+                            min_t_value = min(min_t_value, t_value)
+                    if min_t_value > quantile_1[sample_index]:
+                        current_power += 1
+                current_power /= iteration_size
+            sample_sizes.append(sample_size)
+        return np.max(sample_sizes)
+
+
+# Применение метода
```
Comment (VasilievaKA) [inline hypex/ab_test/limit_distribution.py:132]: Remove tests and usage of this method

File: hypex/ab_test/limit_distribution.py [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+            current_power = 0  # мощность
+            while current_power < 1 - power_level:
+                sample_size += 100
+                current_power = 0
+                total_samples = norm.rvs(size=[iteration_size, number_of_samples], random_state=random_state)
+                for sample in total_samples:
+                    min_t_value = np.inf
+                    for i in range(number_of_samples):
+                        if i != sample_index:
+                            t_value = sample[sample_index] / np.sqrt(1 + variances[i] / variances[sample_index]) - \
```
Comment (VasilievaKA) [inline hypex/ab_test/limit_distribution.py:120]: Dmatryus told me, that \ for line breaks is not a good pattern. Use () for your formula

**Review by VasilievaKA — APPROVED** [review]
Good!

**Comment by tikhomirovd** [comment]
## Objective

In this milestone, we aim to address the challenge of multiple hypothesis testing, focusing on the selection of the optimal sample from a set of 'k' samples based on a specific target metric.

## Hypotheses Formulation

- **Hypothesis H_i**: The 'i-th' sample is the best among the set, where i ranges from 1 to k.
- **Null Hypothesis H_0**: There is no single best sample in the set.

## Methodology

We have developed a method grounded in the theory of limit distributions. This innovative approach is designed to adhere strictly to the predefined probabilities of Type I and Type II errors. By leveraging this method, we can make more accurate and reliable decisions in the context of multiple hypothesis testing.

## Key Features

- **Limit Distribution-Based Method**: Our approach utilizes the concept of limit distributions, offering a robust framework for hypothesis testing.
- **Error Probability Control**: The method ensures strict adherence to the pre-set probabilities of Type I (false positive) and Type II (false negative) errors, enhancing the reliability of the testing process.

## Expected Outcomes

The implementation of this method is expected to significantly improve the accuracy and reliability of multiple hypothesis testing in our project, providing a solid foundation for making data-driven decisions.

**Pull request merged**